src/libsync/pushnotifications.cpp

```diff
@@ -28,7 +28,14 @@ Q_LOGGING_CATEGORY(lcPushNotifications, "nextcloud.sync.pushnotifications", QtIn
 PushNotifications::PushNotifications(Account *account, QObject *parent)
     : QObject(parent)
     , _account(account)
+    , _webSocket(new QWebSocket(QString(), QWebSocketProtocol::VersionLatest, this))
```

Comment:
But now, IIRC you're creating always the websocket even when not used (again IIRC, the design was that the PushNotifications object was always there but not the QWebSocket object). If my memory doesn't betray me, this is a bit wasteful now.

I think you could have just put the two ifs in openWebSocket together. That's where the mistake was, on several calls to it you'd have several connects to the same websocket object hence the multiple slot calls.

Reply:
The push notifications object gets only created if the capabilities report that push notifications are available. See `Account::trySetupPushNotifications()`. With the new reconnect strategy, the push notifications should always try to reconnect if the capabilities allow it, so the websocket needs to stay there forever. Even in the past we could have created the websocket from the beginning, don't know why I did it not. The only difference in the past was, that after three failed connection attempts the websocket could be deleted because it wouldn't be used anymore.

The connections to the webSocket signals were guarded by `Qt::UniqueConnection` and the multiple emitted error signals occurred even with this new version. In my local tests, I had even more than eight error signals right from the beginning. I still find that a little bit strange.

Reply:
> The push notifications object gets only created if the capabilities report that push notifications are available. See `Account::trySetupPushNotifications()`.

Memory failed then ignore me. ;-)